src/visualization/include/iDynTree/Visualizer.h

```diff
@@ -750,6 +750,14 @@ class IModelVisualization
      */
     virtual bool setLinkColor(const LinkIndex& linkIndex, const ColorViz& linkColor) = 0;
 
+    /**
+     * Set the color of a single visual of a given link.
+     *
+     * This will overwrite the material of the visual, but it can be
+     * reset by resetLinkColor.
+     */
+    virtual bool setVisualColor(const LinkIndex& linkIndex, const std::string& visualName, const ColorViz& visualColor) = 0;
```

Comment:
Not problematic, just an observation: for coherence between link and visual whould be cool to have also `setVisualColor(const std::string& linkName, const std::string& visualName, const ColorViz& visualColor)`, but this for sure do not block the PR.

Reply:
Actually, there is no ``SetLinkColor`` that takes the name of the link as argument. That's why I preferred to keep using the ``linkIndex``.

Reply:
Yeah, the index of the visual is a bit opaque, as it depends on the parsing of the model. So, for the scope of the PR, I preferred to keep only this version.

Reply:
Ack, but also the link index depends on the parsing, no?

Reply:
Yes. At least there is a method to get a link index given the name, but there was not a counterpart for the visuals.